### PR TITLE
Megas for All: Add Mega Evolutions from slate 28

### PR DESCRIPTION
**Pull request by HematiteH2O**
* Adds Mega Alolan Ninetales, Mega Zoroark and Mega Delphox and their custom Abilities and Mega Stones
* The Z-Move and Mega Evolution scripts are also modified slightly to account for Zoroark
* Also modifies the stats of Thievul and adds a new custom Ability for it as a buff
* Bans Mega Dragonite (which is a temporary measure decided by the competitive council until a proper nerf is in place)
* Specifies battleOnly for a handful of Mega Evolutions with multiple base forms, because those had odd interactions in a few cases and this fixes them

I hope all of this is right! Let me know if there's anything I need to change! C:

### CODE REVIEW

File: data/mods/megasforall/abilities.ts [hunk 1]
```diff
@@ -1705,9 +1705,250 @@ export const Abilities: {[abilityid: string]: ModdedAbilityData} = {
...
+			pokemon.addVolatile('forgery');
+			let i;
+			for (i = pokemon.side.pokemon.length - 1; i > pokemon.position; i--) {
+				if (!pokemon.side.pokemon[i]) continue;
+				if (pokemon.side.pokemon[i].fainted) {
+					continue;
+				} else {
+					break;
+				}
+			}
```
Comment (KrisXV) [inline data/mods/megasforall/abilities.ts:1844]:
```suggestion
				if (!pokemon.side.pokemon[i] || pokemon.side.pokemon[i].fainted) continue;
				break;
			}
```

File: data/mods/megasforall/abilities.ts [hunk 2]
```diff
@@ -1705,9 +1705,250 @@ export const Abilities: {[abilityid: string]: ModdedAbilityData} = {
...
+		name: "Clairvoyance",
+		rating: 3,
+		num: -1051,
+	},
+	sleightofhand: {
+		desc: "This Pokémon's contact moves become special attacks and do not make contact with the target.",
+		shortDesc: "This Pokémon's contact moves become special and non-contact.",
+		onModifyMove(move) {
+			if (move.flags['contact']) {
+				move.category = 'Special';
```
Comment (KrisXV) [inline data/mods/megasforall/abilities.ts:1941]:
```suggestion
				if (move.category !== 'Special') move.category = 'Special';
```
grass knot, petal dance, etc are already contact

File: data/mods/megasforall/items.ts [hunk 3]
```diff
@@ -1277,4 +1277,47 @@ export const Items: {[itemid: string]: ModdedItemData} = {
...
 	},
+	ninetalesite: {
+		name: "Ninetalesite",
+		spritenum: 578,
+		megaStone: "Ninetales-Alola-Mega",
+		megaEvolves: "Ninetales-Alola",
+		itemUser: ["Ninetales-Alola"],
+		onTakeItem(item, source) {
+			if (source.species.name === 'Ninetales-Alola') return false;
+			if (source.species.name === 'Ninetales-Alola-Mega') return false;
```
Comment (KrisXV) [inline data/mods/megasforall/items.ts:1288]:
```suggestion
			if (source.species.name.startsWith('Ninetales-Alola')) return false;
```

**Review by KrisXV — COMMENTED** [review]
looks good after the suggestions i made!

File: data/mods/megasforall/abilities.ts (hunk 2 above)
Comment (HematiteH2O) [reply data/mods/megasforall/abilities.ts:1937]: Oh, right, good catch!

**Comment by HematiteH2O** [comment]
Thanks so much for the help!!

**Pull request merged**